src/sql/workbench/contrib/dashboard/browser/core/dashboardTab.contribution.ts

```diff
@@ -25,6 +25,7 @@ export interface IDashboardTabContrib {
 	description?: string;
 	alwaysShow?: boolean;
 	isHomeTab?: boolean;
+	isTopLevelTab?: boolean;
```

Comment:
Is adding flags like this sustainable for the long-term development of the project?

Reply:
could you please elaborate more?

Reply:
Essentially at this point it's becoming flag oriented programming. Are we going to keep adding these special case flags?

Reply:
adding a property seems easy and necessary to achieve my goal, what is your recommended way?

Reply:
Well the point I was trying to make is how many of these attributes are we looking to add? Because eventually it's going to be a massive if statement to know where a component ends up going to.

I don't really have a suggestion other than creating a new component that knows how the component itself fits into the dashboard.

Reply:
i see, at this point, i don't see a need of adding more properties, and based on the new design i think i can remove the alwaysShow property. I do agree if we have to many special cases we will need to use classes to handle them.